frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,53 @@
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
-
+export const ConfigMapData = ({data}) => {
   const dl = [];
-  Object.keys(data || []).sort().forEach(k => {
+  Object.keys(data || {}).sort().forEach(k => {
     dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
+    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
   });
 
   return <dl>{dl}</dl>;
 };
 
-export default ConfigMapAndSecretData;
+export class SecretData extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { showSecret: false };
+    this.toggleSecret = this.toggleSecret.bind(this);
+  }
+
+  toggleSecret() {
+    this.setState({ showSecret: !this.state.showSecret });
+  }
+
+  render() {
+    const { data } = this.props;
+    const { showSecret } = this.state;
+    const dl = [];
+    const masked = <React.Fragment>
```

Comment:
Missed a follow-up, my bad, but there is still some duplication here, another `const dl = []` && `dl.push()` of similar markup again.  We should factor that out.

Reply:
Also,`const masked` seems to indicate this could also be a component `<MaskedInput />.  I'd suggest as another follow-on && split this out.

@spadgett @jhadvig, we had a lot of copy/paste duplication in the old console, I think we should be extra vigilant to avoid that if we can this time around.